plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/save-hub.tsx

```diff
@@ -115,7 +124,32 @@ export const SaveHub = () => {
 				__unstableMarkLastChangeAsPersistent();
 			}
 		}
-	};
+	}, [
+		dirtyEntityRecords,
+		editEntityRecord,
+		saveEditedEntityRecord,
+		saveSpecifiedEntityEdits,
+		__unstableMarkLastChangeAsPersistent,
+	] );
+
+	const isMainScreen = urlParams.path === '/customize-store/assembler-hub';
+
+	// Trigger a save when the editor is loaded abd there are unsaved changes in main screen.  This is needed to ensure FE is displayed correctly	because some patterns have dynamic attributes that only generate in Editor.
```

Comment:
```suggestion
	// Trigger a save when the editor is loaded and there are unsaved changes in main screen. This is needed to ensure FE is displayed correctly because some patterns have dynamic attributes that only generate in Editor.
```